### PR TITLE
Add `first` and `last` groupby aggregations support in cudf/dask_cudf

### PR DESCRIPTION
**Pull request by shwina**
Closes https://github.com/rapidsai/cudf/issues/8966

### CODE REVIEW

File: python/cudf/cudf/tests/test_groupby.py [hunk 1]
```diff
@@ -2086,3 +2086,21 @@ def test_groupby_describe(data, group):
...
+        {"a": [], "b": []},
+        {"a": [2, 1, 2, 1, 1, 3], "b": [None, 1, 2, None, 2, None]},
+        {"a": [None], "b": [None]},
+        {"a": [2, 1, 1], "b": [None, 1, 0], "c": [None, 0, 1]},
+    ],
+)
+@pytest.mark.parametrize("agg", ["first", "last", ["first", "last"]])
+def test_groupby_first(data, agg):
+    pdf = pd.DataFrame(data)
+    gdf = cudf.from_pandas(pdf)
```
Comment (skirui-source) [inline python/cudf/cudf/tests/test_groupby.py:2103]: Everything looks good to me! Would only add a test case for a dataframe with  empty and non-empty values.

File: python/dask_cudf/dask_cudf/tests/test_groupby.py [hunk 2]
```diff
@@ -594,3 +594,24 @@ def test_groupby_unique_lists():
     dd.assert_eq(
         gdf.groupby("a").b.unique(), gddf.groupby("a").b.unique().compute(),
     )
+
+
+@pytest.mark.parametrize("agg", ["first", "last"])
+def test_groupby_first_last(agg):
+    pdf = pd.DataFrame(
+        {"a": [2, 1, 2, 1, 1, 3], "b": [None, 1, 2, None, 2, None]}
```
Comment (skirui-source) [inline python/dask_cudf/dask_cudf/tests/test_groupby.py:602]: Add more test cases as in cudf /test_groupby

File: python/cudf/cudf/tests/test_groupby.py (hunk 1 above)
Comment (shwina) [reply python/cudf/cudf/tests/test_groupby.py:2103]: Thanks!

File: python/dask_cudf/dask_cudf/tests/test_groupby.py (hunk 2 above)
Comment (shwina) [reply python/dask_cudf/dask_cudf/tests/test_groupby.py]: Good catch! Added more tests here

**Review by skirui-source — APPROVED** [review]

File: python/dask_cudf/dask_cudf/tests/test_groupby.py [hunk 3]
```diff
@@ -594,3 +594,31 @@ def test_groupby_unique_lists():
...
+        {"a": [], "b": []},
+        {"a": [2, 1, 2, 1, 1, 3], "b": [None, 1, 2, None, 2, None]},
+        {"a": [None], "b": [None]},
+        {"a": [2, 1, 1], "b": [None, 1, 0], "c": [None, 0, 1]},
+    ],
+)
+@pytest.mark.parametrize("agg", ["first", "last"])
+def test_groupby_first_last(data, agg):
+    pdf = pd.DataFrame(data)
+    gdf = cudf.DataFrame.from_pandas(pdf)
```
Comment (rjzamora) [inline python/dask_cudf/dask_cudf/tests/test_groupby.py:611]: This is a pretty nice test, but perhaps it also makes sense to check that the cudf result matches dask-cudf?

Comment (shwina) [reply python/dask_cudf/dask_cudf/tests/test_groupby.py:611]: Done!

**Comment by shwina** [comment]
sigh... got in a bit of git hell. Going to close and open a new PR.

**Pull request closed without merging**